syntax = "proto3";

option java_multiple_files = false;
option java_package = "com.ewolff.microservice.catalog.grpc";
option java_outer_classname = "CatalogProto";

import "google/protobuf/empty.proto";

service CatalogService {
  rpc GetItem (ItemRequest) returns (ItemResponse);
  rpc ListItems (google.protobuf.Empty) returns (ItemListResponse);
}

message ItemRequest {
  int64 itemId = 1;
}

message ItemResponse {
  int64 itemId = 1;
  string name = 2;
  double price = 3;
}

message Item {
  int64 itemId = 1;
  string name = 2;
  double price = 3;
}

message ItemListResponse {
  repeated Item items = 1;
}
